### ISSUE TITLE
Pyright does not report if return type is not declared and contravariant

### ISSUE THREAD

**Issue by juk-r** [microsoft/pyright#6414]
Pyright reports "Contravariant type variable cannot be used in return type" only if it's declared, but not if it's automatically evaluated.
```python
from typing import TypeVar, Generic

T_contra = TypeVar("T_contra", contravariant=True)

class Example(Generic[T_contra]):
    def __init__(self, val: T_contra):
        self.val = val

    def get_val(self): # -> T_contra:
        # no errors in this function
        return self.val
```

**Comment by erictraut** [comment]
This is by design, so I don't consider it a bug. The check for incorrect variance is performed on explicitly annotated parameters and return types.

I think it's a reasonable feature request to extend this to inferred return types, so I'll convert this to an enhancement request.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
